tests/bulk-tx/src/bulk_tx.ts

```diff
@@ -0,0 +1,145 @@
+import { ApiPromise, WsProvider, Keyring } from '@polkadot/api';
+import { SignerResult } from '@polkadot/api/types';
+import type { EventRecord, ExtrinsicStatus, H256, SignedBlock } from '@polkadot/types/interfaces';
+import type { ISubmittableResult, SignatureOptions } from '@polkadot/types/types';
+
+const ALICE = '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY';
+const keyring = new Keyring({ type: 'sr25519' });
+
+//batch size and payload size are set here
+const batch = 3;
+const size = 100;
+
+async function createApi(): Promise<ApiPromise> {
+  // Initialise the provider to connect to the local node
+  // const provider = new WsProvider('wss://polygon-da-explorer.matic.today/ws');
+  const provider = new WsProvider('ws://127.0.0.1:9944');
+
+  // Create the API and wait until ready
+  return ApiPromise.create({
+    provider,
+    types: {
+      DataLookup: {
+        size: 'u32',
+        index: 'Vec<(u32,u32)>'
+      },
+      KateExtrinsicRoot: {
+        hash: 'Hash',
+        commitment: 'Vec<u8>',
+        rows: 'u16',
+        cols: 'u16'
+      },
+      KateHeader: {
+        parentHash: 'Hash',
+        number: 'Compact<BlockNumber>',
+        stateRoot: 'Hash',
+        extrinsicsRoot: 'KateExtrinsicRoot',
+        digest: 'Digest',
+        appDataLookup: 'DataLookup'
+      },
+      Header: 'KateHeader',
+      AppId: 'u32',
+      // CheckAppId: {
+      //     extra: {
+      //         appId: 'u32', 
+      //     },
+      //     types: {}
+      // }
+    },
+    signedExtensions: {
+      CheckAppId: {
+        extrinsic: {
+          appId: 'u32'
+        },
+        payload: {}
+      },
+    },
+  });
+}
+
+//generating random data as per payload specified
+function generateData(size: number): string {
+  let randomChars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+  let result = '';
+  for (let i = 0; i < size; i++) {
+    result += randomChars.charAt(Math.floor(Math.random() * randomChars.length));
+  }
+  return result
+}
+
+async function getNonce(api: ApiPromise, address: any) {
+  const nonce = (await api.rpc.system.accountNextIndex(address)).toNumber();
+  return nonce;
+}
+
+interface SignatureOptionsNew extends SignatureOptions {
+  app_id: number
+}
+
+async function sendTx(api: ApiPromise, sender: any, nonce: any): Promise<any> {
+  try {
+    let data = generateData(size);
+    const nonc = await api.rpc.system.accountNextIndex(sender.address);
```

Comment:
To safely send multiple concurrent tx, we should probably manage nonce on our own.